modules/shortcodes/scribd.php

```diff
@@ -50,8 +50,9 @@ function scribd_shortcode_handler( $atts ) {
  * @param array $atts Shortcode attributes.
  */
 function scribd_shortcode_markup( $atts ) {
-	$markup = <<<EOD
-<iframe class="scribd_iframe_embed" src="//www.scribd.com/embeds/$atts[id]/content?start_page=1&view_mode=$atts[mode]&access_key=$atts[key]" data-auto-height="true" scrolling="no" id="scribd_$atts[id]" width="100%" height="500" frameborder="0"></iframe>
+	$sandbox = class_exists( 'Jetpack_AMP_Support' ) && Jetpack_AMP_Support::is_amp_request() ? 'sandbox="allow-popups allow-scripts allow-same-origin"' : '';
+	$markup  = <<<EOD
+<iframe class="scribd_iframe_embed" src="//www.scribd.com/embeds/$atts[id]/content?start_page=1&view_mode=$atts[mode]&access_key=$atts[key]" $sandbox data-auto-height="true" scrolling="no" id="scribd_$atts[id]" width="100%" height="500" frameborder="0"></iframe>
```

Comment:
Just like for Google Maps, should we take that opportunity to clean things up a bit? We could even make that string translatable, switch everyone to HTTPS, and ensure that link opening in a new window has all the right attributes.

```diff
diff --git a/modules/shortcodes/scribd.php b/modules/shortcodes/scribd.php
index 4cba34662..7bf149fbe 100644
--- a/modules/shortcodes/scribd.php
+++ b/modules/shortcodes/scribd.php
@@ -50,13 +50,24 @@ function scribd_shortcode_handler( $atts ) {
  * @param array $atts Shortcode attributes.
  */
 function scribd_shortcode_markup( $atts ) {
-	$sandbox = class_exists( 'Jetpack_AMP_Support' ) && Jetpack_AMP_Support::is_amp_request() ? 'sandbox="allow-popups allow-scripts allow-same-origin"' : '';
-	$markup  = <<<EOD
-<iframe class="scribd_iframe_embed" src="//www.scribd.com/embeds/$atts[id]/content?start_page=1&view_mode=$atts[mode]&access_key=$atts[key]" $sandbox data-auto-height="true" scrolling="no" id="scribd_$atts[id]" width="100%" height="500" frameborder="0"></iframe>
-<div style="font-size:10px;text-align:center;width:100%"><a href="https://www.scribd.com/doc/$atts[id]" target="_blank">View this document on Scribd</a></div>
-EOD;
+	$sandbox = class_exists( 'Jetpack_AMP_Support' ) && Jetpack_AMP_Support::is_amp_request()
+		? 'sandbox="allow-popups allow-scripts allow-same-origin"'
+		: '';
+	$url     = sprintf(
+		'https://www.scribd.com/embeds/%1$d/content?start_page=1&view_mode=%2$s&access_key=%3$s',
+		absint( $atts['id'] ),
+		esc_attr( $atts['mode'] ),
+		esc_attr( $atts['key'] )
+	);
 
-	return $markup;
+	return sprintf(
+		'<iframe class="scribd_iframe_embed" src="%1$s" %2$s data-auto-height="true" scrolling="no" id="scribd_%3$d" width="100%%" height="500" frameborder="0"></iframe>
+		<div style="font-size:10px;text-align:center;width:100%%"><a href="https://www.scribd.com/doc/%3$d" rel="noopener noreferrer" target="_blank">%4$s</a></div>',
+		esc_url( $url ),
+		$sandbox,
+		absint( $atts['id'] ),
+		esc_html__( 'View this document on Scribd', 'jetpack' )
+	);
 }
 add_shortcode( 'scribd', 'scribd_shortcode_handler' );
 

```

Reply:
Good idea, Jeremy! If it's alright, I'll commit this later tonight.

Reply:
How does https://github.com/Automattic/jetpack/pull/15422/commits/1d847b04459faa4e8be6458a5d3b051fddff84b9 look?